### PR TITLE
services/horizon/internal/db2/schema: Replace https://github.com/jteeuwen/go-bindata dependency

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### Summary

Replace https://github.com/jteeuwen/go-bindata dependency

### Goal and scope

We should not be using https://github.com/jteeuwen/go-bindata for the following reasons:

* The library is no longer maintained 
* The code generated by `go-bindata` is not formatted properly
* The code generated by `go-bindata` can produce `go vet` / `golint` warnings

### Summary of changes

* update notes_for_developers.md
* add tests which ensure the generated binary assets match the sql files in the `assets` directory. this is useful because it will prevent mistakes where someone has added or updated a migration file but forgot to run the script to generate the binary data

### Known limitations & issues

Unfortunately we still need to check in the generated code into the repo. I did look into using https://github.com/gobuffalo/packr . With packr you don't need to include the generated code in the git repo. However, if we were to use packr we wouldn't be able to use `go get` or `go build` to fetch / build the horizon binary. Instead we'd need to use packr's toolchain    

### What shouldn't be reviewed

N / A

### CODE REVIEW

**Comment by kevinburke** [comment]
Hey, definitely, I fixed all of the lint/formatting issues and for what it's worth I'm committed to maintaining the package/responding to bug reports etc.

File: services/ticker/internal/tickerdb/migrations/bindata.go [hunk 1]
```diff
@@ -75,436 +71,236 @@ func (fi bindataFileInfo) Mode() os.FileMode {
...
...10\x4f\x65\x1a\x57\x29\x54\xf1\xa7\x2c\x85\x61\x6c\x0c\xe9\xb7\xca\x7b\x0c\x1e\x22\x01\x00\x40\x2d\x78\x74\xa4\x0c\xe4\x45\x05\x79\x9d\x65\xf0\x5c\x6e\xbe\xc6\xe5\x16\xbe\xa4\xdb\xfb\x85\xd1\xdc\x22\xe8\x4e\x39\xa5\x03\x3a\xd8\x2b\x37\x91\xdd\x45\xef\xde\xdf\xfd\x15\x1d\x41\xf2\x7e\x44\x07\x01\x7f\x86\x8b\x93\x30\x0d\x6b\x16\x1f\x3f\x5c\x5a\xd8\xb1\x97\x4a\x6b\x1e\x6d\xf0\x40\x36\xe0\x0e\xdd\x05\xa2\xc6\xd0\x49\x87\x3f\x46\x72\xd8\x42\xc3\x6c\x50\xd9\x75\x66\xcf\x5a\x35\x06\xaf\x41\xfd\x3c\x06\x5a\x1e\x67\x66\x70\xa8\xc9\x13\xff\x47\xcd\x71\x49\xcd\x36\x38\x36\x06\x5b\xd9\x4c\xb2\xe5\x5e\x91\xbd\x66\x6b\x75\xc7\x4e\xfe\xd1\xdd\x98\xdd\x99\xea\xe6\xb8\xc8\xcb\xbd\x32\x74\x2d\x86\xe5\x4c\x05\x62\x2b\xd1\x39\x5e\x7d\x1a\xa3\x7c\x38\x99\x04\xea\xd1\x07\xd5\x0f\x70\xa0\xd0\x2d\x9f\xf0\x8b\x2d\xae\x29\x74\x87\xfa\x05\x5f\xd1\x24\xe9\xe7\xb8\xce\x2a\xb0\x7c\x88\xfe\x5d\x5b\xdc\x3d\x0a\x11\x67\x55\x5a\x9e\x4a\x59\xe4\xd9\xf6\xbc\x99\xcb\x8c\x38\x49\xe0\xa9\xc8\xbf\x57\x65\xbc\xc9\xab\xe3\x23\xf8\x25\x4d\x79\x6c\x99\x7c\xc1\x09\xea\x7c\xf3\xad\x4e\x21\x9a\xff\xdf\x9f\xea\x37\x0f\x38\xdb\x83\x84\x0f\x56\x88\xa4\x2c\x9e\xd7\xf6\xe0\x51\xfc\x0e\x00\x00\xff\xff\x48\x16\x89\x51\x35\x03\x00\x00")
```
[the remark this answers is not in the document]
Comment (tamirms) [reply services/ticker/internal/tickerdb/migrations/bindata.go]: The old bindata did not match file on disk.

`20190404184050-initial.sql` on disk ends the drop table statement with a semicolon: `DROP TABLE public.assets;`. The file included in the old bindata omitted the semicolon: `DROP TABLE public.assets`

File: services/horizon/internal/docs/notes_for_developers.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Compile and install Horizon as described in the [Horizon development guide](deve
 ## <a name="regen"></a> Regenerating generated code
 
 Horizon uses two Go tools you'll need to install:
-1. [go-bindata](https://github.com/jteeuwen/go-bindata) is used to bundle test data
+1. [go-bindata](github.com/kevinburke/go-bindata) is used to bundle test data
```
Comment (bartekn) [inline services/horizon/internal/docs/notes_for_developers.md]: Do we actually use it anywhere?

Comment (tamirms) [reply services/horizon/internal/docs/notes_for_developers.md]: in `rebuild_schema.bash` it executes the following commands:

```
go generate github.com/stellar/go/services/horizon/internal/db2/schema
go generate github.com/stellar/go/services/horizon/internal/test
```

These commands will only work if you have `github.com/kevinburke/go-bindata` installed

Comment (bartekn) [reply services/horizon/internal/docs/notes_for_developers.md]: I see, I'm wondering if we should add it to dependencies and run using `go run ...`. It's easy to forget or use a different version.

**Pull request merged**